ibm/resource_ibm_tg_gateway.go

```diff
@@ -0,0 +1,417 @@
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Second,
+	}
+
+	return stateConf.WaitForState()
+}
+
+func isTransitGatewayRefreshFunc(client *transitgatewayapisv1.TransitGatewayApIsV1, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		gettgwoptions := &transitgatewayapisv1.DetailTransitGatewayOptions{
+			ID: &id,
+		}
+		transitGateway, response, err := client.DetailTransitGateway(gettgwoptions)
+		if err != nil {
+			return nil, "", fmt.Errorf("Error Getting Transit Gateway: %s\n%s", err, response)
+		}
+
+		if *transitGateway.Status == "available" || *transitGateway.Status == "failed" {
+			return transitGateway, isTransitGatewayProvisioningDone, nil
+		}
+
+		return transitGateway, isTransitGatewayProvisioning, nil
+	}
+}
+
+func resourceIBMTransitGatewayRead(d *schema.ResourceData, meta interface{}) error {
+	ID := d.Id()
+	return transitGatewayGet(d, meta, ID)
+}
+
+func transitGatewayGet(d *schema.ResourceData, meta interface{}, id string) error {
+	client, err := transitgatewayClient(meta)
+	if err != nil {
+		return err
+	}
+	tgOptions := &transitgatewayapisv1.DetailTransitGatewayOptions{}
+	if id != "" {
+		tgOptions.ID = &id
+	}
+
+	log.Println("Inside get detail tgw now")
+	tgw, _, err := client.DetailTransitGateway(tgOptions)
+	if err != nil {
+		return err
+	}
+
+	log.Println("Got detail tgw now")
+	d.SetId(*tgw.ID)
+	d.Set("crn", tgw.Crn)
+	d.Set("name", tgw.Name)
+	d.Set("location", tgw.Location)
+	d.Set("created_at", tgw.CreatedAt.String())
+
+	log.Println("Updated Field is:", tgw.UpdatedAt)
+	if tgw.UpdatedAt != nil {
+		d.Set("updated_at", tgw.UpdatedAt.String())
+	}
+	d.Set("global", tgw.Global)
+	d.Set("status", tgw.Status)
+
+	tags, err := GetTagsUsingCRN(meta, *tgw.Crn)
+	if err != nil {
+		log.Printf(
+			"Error on get of transit gateway (%s) tags: %s", d.Id(), err)
+	}
+	d.Set(tgGatewayTags, tags)
+
+	controller, err := getBaseController(meta)
+	if err != nil {
+		return err
+	}
+
+	d.Set(ResourceControllerURL, controller+"/interconnectivity/transit")
+	d.Set(ResourceName, *tgw.Name)
+	d.Set(ResourceCRN, *tgw.Crn)
+	d.Set(ResourceStatus, *tgw.Status)
+	if tgw.ResourceGroup != nil {
+		rg := tgw.ResourceGroup
+		d.Set("resource_group", *rg.ID)
+		d.Set(ResourceGroupName, *rg.ID)
+	}
+	return nil
+}
+
+func resourceIBMTransitGatewayUpdate(d *schema.ResourceData, meta interface{}) error {
+	client, err := transitgatewayClient(meta)
+
+	if err != nil {
+		return err
+	}
+
+	ID := d.Id()
+	tgOptions := &transitgatewayapisv1.DetailTransitGatewayOptions{
+		ID: &ID,
+	}
+	tgw, resp, err := client.DetailTransitGateway(tgOptions)
+
+	if err != nil {
+		log.Printf("Error fetching Tranisit  Gateway: %s", resp)
+		return err
+	}
+
+	updateTransitGatewayOptions := &transitgatewayapisv1.UpdateTransitGatewayOptions{}
+	updateTransitGatewayOptions.ID = &ID
+	if d.HasChange(tgName) {
+		if _, ok := d.GetOk(tgName); ok {
```

Comment:
no need to call two times d.Get(tgName)

replace _ with some variable and use it

Reply:
ok changed.